client/my-sites/domains/domain-management/settings/set-as-primary/index.tsx

```diff
@@ -0,0 +1,151 @@
+	showUpdatePrimaryDomainErrorNotice,
+} from 'calypso/state/domains/management/actions';
+import { getCurrentRoute } from 'calypso/state/selectors/get-current-route';
+import hasActiveSiteFeature from 'calypso/state/selectors/has-active-site-feature';
+import isDomainOnlySite from 'calypso/state/selectors/is-domain-only-site';
+import { setPrimaryDomain } from 'calypso/state/sites/domains/actions';
+import type {
+	SetAsPrimaryProps,
+	SetAsPrimaryPassedProps,
+	SetAsPrimaryStateProps,
+	ChangePrimaryFunctionSignature,
+} from './types';
+
+const SetAsPrimary = ( props: SetAsPrimaryProps ): JSX.Element | null => {
+	const translate = useTranslate();
+	const [ isSettingPrimaryDomain, setIsSettingPrimaryDomain ] = useState( false );
+	const {
+		domain,
+		hasDomainOnlySite,
+		isOnFreePlan,
+		hasNonPrimaryDomainsFlag,
+		canSetPrimaryDomain,
+		isManagingAllSites,
+		selectedSite,
+	} = props;
+
+	const shouldUpgradeToMakeDomainPrimary = () => {
+		return (
+			hasNonPrimaryDomainsFlag &&
+			isOnFreePlan &&
+			( domain.type === type.REGISTERED || domain.type === type.MAPPED ) &&
+			! hasDomainOnlySite &&
+			! domain.isPrimary &&
+			! domain.isWPCOMDomain &&
+			! domain.isWpcomStagingDomain &&
+			! canSetPrimaryDomain
+		);
+	};
+
+	const canSetAsPrimary = () => {
+		return (
+			! isManagingAllSites &&
+			domain &&
+			domain.canSetAsPrimary &&
+			! domain.isPrimary &&
+			! shouldUpgradeToMakeDomainPrimary()
+		);
+	};
+
+	const setPrimaryDomain = ( domainName: string ) => {
+		return new Promise( ( resolve, reject ) => {
+			props.setPrimaryDomain( selectedSite.ID, domainName, ( error, data ) => {
+				if ( ! error && data && data.success ) {
+					resolve( null );
+				} else {
+					reject( error );
+				}
+			} );
+		} );
+	};
+
+	const handleSetPrimaryDomainClick = async () => {
+		setIsSettingPrimaryDomain( true );
+		props.changePrimary( domain, 'button_click' );
+		try {
+			await setPrimaryDomain( domain.name );
+			props.showUpdatePrimaryDomainSuccessNotice( domain.name );
+		} catch ( error ) {
+			props.showUpdatePrimaryDomainSuccessNotice( domain.name );
+		} finally {
+			setIsSettingPrimaryDomain( false );
+		}
+	};
+	if ( ! canSetAsPrimary() ) return null;
+	return (
+		<div className="set-as-primary">
+			<Accordion
+				title={ translate( 'Set as primary' ) }
+				subtitle={ translate( 'Make this domain your primary site address' ) }
+			>
+				<p className="set-as-primary__content">
+					{ translate( 'Your current primary site address is %(domainName)s', {
+						args: {
+							domainName: selectedSite.domain,
+						},
+					} ) }
+				</p>
+				<Button onClick={ handleSetPrimaryDomainClick } busy={ isSettingPrimaryDomain }>
+					{ translate( 'Set this domain as primary' ) }
+				</Button>
+			</Accordion>
+		</div>
+	);
+};
+
+const changePrimary: ChangePrimaryFunctionSignature = ( domain, mode ) =>
+	composeAnalytics(
+		recordGoogleEvent(
+			'Domain Management',
+			'Changed Primary Domain to in Settings',
```

Comment:
I think the `to` in the original message was a typo, this could be written as `Changed Primary Domain in Settings` 🙂

Reply:
You right, I totally forgot to update it 😅